### PR TITLE
[juce] update Juce to 8.0.4

### PR DESCRIPTION
**Pull request by aminya**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
From arm64-windows config log:
~~~
-- Configuring juceaide
-- Building juceaide
-- Exporting juceaide
-- Testing juceaide
CMake Error at extras/Build/juceaide/CMakeLists.txt:191 (message):
  Testing juceaide failed:
~~~
This port doesn't have an arm windows problem. This port has a cross-build problem.
Upstream tries to reinvoke CMake to build host tools.
The vcpkg port must make available and use the tools from the host triplet instead.

**Comment by dg0yt** [comment]
https://github.com/juce-framework/JUCE/blob/51d11a2be6d5c97ccf12b4e5e827006e19f0555a/extras/Build/juceaide/CMakeLists.txt#L33-L36

File: ports/juce/0002-build-linux-find_packages.patch [hunk 1]
```diff
@@ -2,35 +2,36 @@ diff --git a/extras/Build/CMake/JUCEModuleSupport.cmake b/extras/Build/CMake/JUC
 index e18dd5e08..45cc9245f 100644
 --- a/extras/Build/CMake/JUCEModuleSupport.cmake
 +++ b/extras/Build/CMake/JUCEModuleSupport.cmake
-@@ -353,6 +353,29 @@ function(_juce_create_pkgconfig_target name)
```
Comment (BillyONeal) [inline ports/juce/0002-build-linux-find_packages.patch:5]: I believe this is mostly a line endings change, but the PR changes it to the correct (LF)s

**Comment by BillyONeal** [comment]
Thanks for the update!

**Comment by aminya** [comment]
> From arm64-windows config log:
> 
> ```
> -- Configuring juceaide
> -- Building juceaide
> -- Exporting juceaide
> -- Testing juceaide
> CMake Error at extras/Build/juceaide/CMakeLists.txt:191 (message):
>   Testing juceaide failed:
> ```
> 
> This port doesn't have an arm windows problem. This port has a cross-build problem. Upstream tries to reinvoke CMake to build host tools. The vcpkg port must make available and use the tools from the host triplet instead.

@dg0yt How do we fix that?

File: ports/juce/0002-build-linux-find_packages.patch (hunk 1 above)
Comment (aminya) [reply ports/juce/0002-build-linux-find_packages.patch:5]: It failed to apply the patch, so I recreate it from the actual code.

**Comment by BillyONeal** [comment]
> Thanks for the update!

I merged this despite the problems enumerated above because it didn't appear to be a regression vs. the status quo.

> @dg0yt How do we fix that? juceaide is a tool used during plugin generation, so it should work for a full working Juce.

I think `JUCE_BUILD_HELPER_TOOLS` should be guarded to only run for the host triplet, and non-host triplets need to get those tools from the host triplet rather than attempting to build them.

**Pull request merged**